pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -147,13 +143,53 @@ func validateFnConfigPath(p string) error {
 	return nil
 }
 
+// GetValidatedFnConfigFromPath validates the functionConfig at the path specified by
+// the package path (pkgPath) and configPath, returning the functionConfig as an
+// RNode if the validation is successful.
+func GetValidatedFnConfigFromPath(pkgPath types.UniquePath, configPath string) (*yaml.RNode, error) {
+	path := filepath.Join(string(pkgPath), configPath)
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("functionConfig must exist in the current package")
+	}
+	reader := kio.ByteReader{Reader: file}
+	nodes, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read functionConfig '%q': %w", configPath, err)
```

Comment:
We don't need quotes around %q. %q quotes the value. https://golang.org/pkg/fmt/